pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/AndOperatorTransformFunction.java

```diff
@@ -45,4 +48,37 @@ int getLogicalFuncResult(int left, int right) {
     }
     return 0;
   }
+
+  @Nullable
+  @Override
+  public RoaringBitmap getNullBitmap(ValueBlock valueBlock) {
+    int numDocs = valueBlock.getNumDocs();
+    int numArguments = _arguments.size();
+    int[][] intValuesSVs = new int[numArguments][numDocs];
+    RoaringBitmap[] nullBitmaps = new RoaringBitmap[numArguments];
+    RoaringBitmap nullBitmap = new RoaringBitmap();
+    for (int i = 0; i < numArguments; i++) {
+      intValuesSVs[i] = _arguments.get(i).transformToIntValuesSV(valueBlock);
+      nullBitmaps[i] = _arguments.get(i).getNullBitmap(valueBlock);
+    }
+    for (int docId = 0; docId < numDocs; docId++) {
```

Comment:
I'm not sure if it is possible, but performance wise, I think it would be better to iterate over the arguments first in order to always use the same nullBitmap and therefore reduce the memory preasure.

Reply:
Done.

You are right. Thank you for the catch.

Reply:
Even better, we could use the roaring bitmap iterator, which is going to be faster than sequential calls to `contains`

Reply:
Even better, we could use the roaring bitmap iterator, which is going to be faster than sequential calls to contains

Reply:
Please ignore my previous comment. It was replied to a wrong comment.

Reply:
Done.

Good suggestion!